src/global/tracking/TrackPropagation_factory.cc

```diff
@@ -0,0 +1,183 @@
+// Created by Tyler Kutz
+// Subject to the terms in the LICENSE file found in the top-level directory.
+
+#include <JANA/JEvent.h>
+#include "TrackPropagation_factory.h"
+#include "algorithms/tracking/JugTrack/TrackingResultTrajectory.hpp"
+#include "services/geometry/acts/ACTSGeo_service.h"
+
+#include <Acts/EventData/MultiTrajectoryHelpers.hpp>
+#include <Acts/Surfaces/RadialBounds.hpp>
+
+#include <extensions/spdlog/SpdlogExtensions.h>
+#include <services/log/Log_service.h>
+#include <spdlog/fmt/ostr.h>
+
+#include <edm4eic/TrackPoint.h>
+#include <edm4eic/TrackSegment.h>
+
+void eicrecon::TrackPropagation_factory::Init() {
+
+	auto app = GetApplication();
+
+	// SpdlogMixin logger initialization, sets m_log
+	InitLogger(app, GetTag());
+
+	m_input_tag = "CentralCKFTrajectories";
+
+	auto acts_service = GetApplication()->GetService<ACTSGeo_service>();
+	m_track_propagation_algo.init(acts_service->actsGeoProvider(), logger());
+
+        m_geoSvc = app->template GetService<JDD4hep_service>();
+
+	SetPropagationSurfaces();
+
+}
+
+void eicrecon::TrackPropagation_factory::ChangeRun(const std::shared_ptr<const JEvent> &event) {
+	JFactoryT::ChangeRun(event);
+}
+
+void eicrecon::TrackPropagation_factory::Process(const std::shared_ptr<const JEvent> &event) {
+
+	auto trajectories = event->Get<eicrecon::TrackingResultTrajectory>(m_input_tag);
+
+	edm4eic::TrackSegmentCollection propagated_tracks;
+
+	for(auto traj: trajectories) {
+		edm4eic::MutableTrackSegment this_propagated_track;
+		for(unsigned short isurf = 0; auto surf: m_target_surface_list) {
+			try {
+				auto prop_point = m_track_propagation_algo.propagate(traj, surf);
+				if(!prop_point) continue;
+				prop_point->surface = m_target_surface_ID[isurf];
+				this_propagated_track.addToPoints(*prop_point);
+			} catch (std::exception& e) {
+				m_log->warn("Exception in underlying algorithm: {}. Trajectory is skipped", e.what());
```

Comment:
Is this branch actually exercised? If not, please remove in case we have exceptions that we want to catch. Blanket-ignoring is a practice that occurs in the existing code, but we are looking to move away from it.

Reply:
Yeah I copied this from another factory but if we're moving away from it I will take it out.